### PR TITLE
Clean redundant DNS records

### PR DESCRIPTION
**Pull request by averdagu**
For each replica of each DB service (NB/SB) ovn-operator will create a DNSData CR. This commit modifies this and each DB service will have each own DNSData CR (ovnsb-cluster, ovnnb-cluster) and this CR will contain all the each pod IP.

Jira: [OSPRH-3628](https://issues.redhat.com//browse/OSPRH-3628)

### CODE REVIEW

File: api/v1beta1/ovndbcluster_types.go [hunk 1]
```diff
@@ -34,6 +34,10 @@ const (
 	// DNSSuffix : hardcoded value on how DNSCore domain is configured
 	DNSSuffix = "cluster.local"
 	// TODO: retrieve it from environment
+	// NBDBDnsName - Northbound DNS object name
+	NBDBDnsName = "ovnnb-cluster"
```
Comment (booxter) [inline api/v1beta1/ovndbcluster_types.go:38]: DNS: https://google.github.io/styleguide/go/decisions#initialisms should retain the original upper case for abbreviations (unless it's unexported and at the start of the variable name, so e.g. dnsList is fine for a variable).

**Review by booxter — CHANGES_REQUESTED** [review]
since you have to rebase, please also change the variable names; but otherwise I think it's good to go

File: controllers/ovndbcluster_controller.go [hunk 2]
```diff
@@ -760,23 +755,28 @@ func (r *OVNDBClusterReconciler) reconcileServices(
...
+			dnsDataName = ovnv1.NBDBDNSName
+		}
+		// Create DNSData CR
+		err = ovndbcluster.DNSData(
+			ctx,
+			helper,
+			serviceName,
+			dnsIPsList,
+			instance,
+			dnsDataName,
```
Comment (karelyatin) [inline controllers/ovndbcluster_controller.go:775]: Instead of dnsDataName i think we can just use serviceName and avoid all the logic above.

**Comment by booxter** [comment]
recheck

**Comment by averdagu** [comment]
recheck

**Comment by karelyatin** [comment]
recheck cert-manager issue fixed

**Review by karelyatin — APPROVED** [review]
/lgtm

**Pull request merged**